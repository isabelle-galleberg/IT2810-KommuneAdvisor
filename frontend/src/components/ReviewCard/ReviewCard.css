.reviewCard {
  background-color: white;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  border: 1px solid #3a73c5;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  padding: 5px 10px;
  width: clamp(200px, calc(75vw), 550px);
}

.reviewCardTop {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.reviewCardBottom {
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.title {
  font-weight: bold;
}

.description {
  font-size: 14px;
}

.date,
.name {
  font-size: 12px;
  color: #6e6e6e;
}
