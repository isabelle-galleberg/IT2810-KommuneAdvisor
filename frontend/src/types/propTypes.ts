export interface KommuneCardProps {
  id: string;
  name: string;
  county: string;
  weaponImg: string;
  rating: number;
}
