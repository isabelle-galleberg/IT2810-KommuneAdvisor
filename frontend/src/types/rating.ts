export interface Rating {
    name: string;
    rating: Number;
    title: string;
    description: string;
    timestamp: Date;
    kommuneNumber: string;
  }