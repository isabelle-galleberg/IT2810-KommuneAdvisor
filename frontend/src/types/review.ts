export interface Review {
  _id?: string;
  title: string;
  description: string;
  rating: number;
  name: string;
  timestamp?: string;
}
