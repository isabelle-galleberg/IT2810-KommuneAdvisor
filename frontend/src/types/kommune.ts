import { Rating } from "./rating";

export interface Kommune {
  kommuneNumber: string;
  name: string;
  population: number;
  areaInSquareKm: number;
  mapUrl: string;
  logoUrl: string;
  writtenLanguage: string;
  county: string;
  kommuneRating: Rating[];
}