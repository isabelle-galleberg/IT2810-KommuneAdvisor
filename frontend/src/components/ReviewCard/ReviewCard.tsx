import './ReviewCard.css';
import { Rating } from 'react-simple-star-rating';
import { Review } from '../../types/review';

export default function ReviewCard({
  title,
  description,
  rating,
  name,
  timestamp,
}: Review) {

  // convert from timestamp to date
  const date = new Date(Number(timestamp)).toLocaleString().split(',')[0];

  return (
    <div className='reviewCard'>
      <div className='reviewCardTop'>
        <Rating
          initialValue={rating}
          readonly
          size={20}
        />
        <div className='date'>{date}</div>
      </div>
      <div className='reviewCardBottom'>
        <div className='title'>{title}</div>
        <div className='description'>{description}</div>
        <div className='name'>{`- ${name}`}</div>
      </div>
    </div>
  );
}
