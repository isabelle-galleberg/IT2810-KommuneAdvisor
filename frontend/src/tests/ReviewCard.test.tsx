import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Review } from '../types/review';
import ReviewCard from '../components/ReviewCard/ReviewCard';

const data: Review = {
  title: 'Fint sted',
  description: 'Molde var ikke s√• stygt som jeg trodde!',
  rating: 4.5,
  name: 'Kong Olav',
};

describe('<ReviewCard />', () => {
  it('should match snapshot with data', () => {
    const elem = renderer
      .create(
        <Router>
          <ReviewCard
            title={data.title}
            description={data.description}
            rating={data.rating}
            name={data.name}
          />
        </Router>
      )
      .toJSON();

    expect(elem).toMatchSnapshot();
  });
});
