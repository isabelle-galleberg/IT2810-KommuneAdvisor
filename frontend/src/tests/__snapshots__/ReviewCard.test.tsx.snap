// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> should match snapshot with data 1`] = `
<div
  className="reviewCard"
>
  <div
    className="reviewCardTop"
  >
    <span
      className="style-module_starRatingWrap__q-lJC"
      style={
        Object {
          "direction": "ltr",
        }
      }
    >
      <span
        aria-hidden="true"
        className="style-module_simpleStarRating__nWUxf react-simple-star-rating"
        style={
          Object {
            "cursor": "",
          }
        }
      >
        <span
          className="style-module_emptyIcons__Bg-FZ empty-icons"
          style={
            Object {
              "color": "#cccccc",
            }
          }
        >
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="style-module_fillIcons__6---A filled-icons"
          style={
            Object {
              "color": "#ffbc0b",
              "left": 0,
              "transition": "",
              "width": "100%",
            }
          }
          title="5 out of 5"
        >
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth={0}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <div
      className="date"
    />
  </div>
  <div
    className="reviewCardBottom"
  >
    <div
      className="title"
    >
      Fint sted
    </div>
    <div
      className="description"
    >
      Molde var ikke s√• stygt som jeg trodde!
    </div>
    <div
      className="name"
    >
      - Kong Olav
    </div>
  </div>
</div>
`;
